components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -2569,6 +2593,28 @@ private ClaimConfig getClaimConfiguration(int applicationId, Connection connecti
             IdentityApplicationManagementUtil.closeResultSet(loadClaimConfigsResultSet);
         }
 
+        PreparedStatement loadSPDialectsPrepStmt = null;
+        ResultSet loadSPDialectsResultSet = null;
+
+        try {
+            loadSPDialectsPrepStmt = connection
+                    .prepareStatement(ApplicationMgtDBQueries.LOAD_SP_DIALECTS_BY_APP_ID);
+            loadSPDialectsPrepStmt.setInt(1, tenantID);
+            loadSPDialectsPrepStmt.setInt(2, applicationId);
+            loadSPDialectsResultSet = loadSPDialectsPrepStmt.executeQuery();
+
+            while (loadSPDialectsResultSet.next()) {
```

Comment:
while and if can be merged, 
while (loadSPDialectsResultSet.next() && loadSPDialectsResultSet.getString(1) != null) {